### DISCUSSION TITLE
aws ec2 describe-instances --filters flag passed with empty string returns instances unfiltered

### DISCUSSION THREAD

**Discussion by SztivT** [aws/aws-cli#8113] [Q&A]
### The problem
When i am passing empty `--filters` option to a `describe-instances` call, it returns with every instance from the region unfiltered, same as with the call not using the option.
The [documentation](https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-instances.html) mentions that the option is not required, but i think passing this flag is intentional, thereby the call should return with empty object, or should fail if no valid filter have been used.

Is this behavior intentional? 

#### The call:
```aws ec2 describe-instances --filters --output json```

#### Expected output
```An error occurred (InvalidOptionValue) when calling the DescribeInstances operation: The  option `filters` provided with zero parameters.```
or

```{}```

#### Current output
```aws ec2 describe-instances --output json```

**Comment by RyanFitzSimmonsAK** [comment]
Hi @SztivT, thanks for reaching out.  Filtering with `--filters` or `--filter` is server-side filtering that can behave differently on a service by service basis.  For example `--filter` while using `aws ce get-cost and usage` does require specifying a filter, and can't just be empty.  However, `aws ec2 describe-instances` does not have this limitation, and it is actually mentioned in the docs.

> If you do not specify instance IDs or filters, the output includes information for all instances, which can affect performance.

This behavior seems to be fairly consistent between services that use `--filters` (not `--filter`).  Could you clarify what you mean by "passing this flag is intentional", and maybe tell me a bit more about your use case?  Thanks!

**Comment by SztivT** [comment]
Hi @RyanFitzSimmonsAK, thanks for the explanation.

By "passing this flag is intentional" i meant using the option in the command presumably not happens by mistake. Beside the error handling i would like to rely on the output without further content checking, that's why I think using the option unparameterized should lead to an error.



### Use case
#### Problem
Cancellation of automation workflow not respects termination of processes which leads to untracked instances with a tag set which is created dynamically.

#### Resolution
Based on the above mentioned tags, an unattended clean up process looks for instance ids with ```aws ec2 describe-instances --filters $filters --output json``` and initiate termination on retrieved instance ids.

**Comment by RyanFitzSimmonsAK** [comment]
Hi @SztivT, thanks for following up.  As I mentioned, the behavior of the `--filters` command is determined by the service team, so I will pass this feedback along to the service team, and provide any updates here.  Thanks!
